schemaregistry/resource_schema.go

```diff
@@ -44,9 +68,32 @@ func resourceSchema() *schema.Resource {
 				Required:    true,
 				Description: "The schema string",
 				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
-					newJSON, _ := structure.NormalizeJsonString(new)
-					oldJSON, _ := structure.NormalizeJsonString(old)
-					return newJSON == oldJSON
+					var schemaEquals bool
+
+					if schemaTypeStr, ok := d.Get("schema_type").(string); ok {
+						if strings.ToLower(schemaTypeStr) == "json" {
+							newJSON, _ := structure.NormalizeJsonString(new)
+							oldJSON, _ := structure.NormalizeJsonString(old)
+							schemaEquals = newJSON == oldJSON
+						} else if strings.ToLower(schemaTypeStr) == "avro" {
+							newJSON, _ := structure.NormalizeJsonString(new)
+							oldJSON, _ := structure.NormalizeJsonString(old)
+							schemaEquals = newJSON == oldJSON
+						} else if strings.ToLower(schemaTypeStr) == "protobuf" {
+							newProtoString, err := FormatProtobufString(new)
```

Comment:
Not sure what to do if there is an error while formatting. Ideally i would propagate an error but it doesnt seem like the CustomizeDiff function accepts an error as a return.